compiler/aarch64/codegen/BinaryEvaluator.cpp

```diff
@@ -30,6 +30,113 @@
 #include "il/Node_inlines.hpp"
 #include "infra/Bit.hpp"
 
+/**
+ * Helper that evaluates a child node.
+ * @param[in] childReg : the register of the child node being looked at
+ * @param[in] trgReg : the target register
+ * @param[in] child : the child node being looked at
+ * @param[in] parent : the parent of the child node being looked at
+ * @param[in] cg : codegenerator
+ */
+static inline void
+evaluateRegister(TR::Register **childReg, TR::Register *trgReg, TR::Node *child, TR::Node *parent, TR::CodeGenerator *cg)
+   {
+   int64_t value;
+	if (child->getOpCode().isLoadConst() && child->getRegister() == NULL)
+      {
+      value = child->getLongInt();
+      //if(constantIsUnsignedImm12(value))
+         //{
+         //TODO: Cover special cases for immediate loads
+         //}
+      //else
+         //{
+         *childReg = cg->allocateRegister();
+         loadConstant64(cg, parent, value, *childReg);
```

Comment:
I think these two statements should just be: `*childReg = cg->evaluate(child)`.  It should do the same thing.

Reply:
Does calling evaluate() handle the constant special case more easily than doing what's there? (I put this in because it follows the idea of the others, and because I thought it was meant to simplify things.)